rca/static_src/javascript/programmes/components/SearchForm.js

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import debounce from 'lodash.debounce';
+
+import {
+    setSearchQuery,
+    clearSearchQuery,
+    searchProgrammes,
+} from '../programmes.slice';
+
+/**
+ * A search form for programmes, visually only appearing as a single field.
+ */
+const SearchForm = ({
+    searchQuery,
+    label,
+    setQuery,
+    clearQuery,
+    startSearch,
+    hasResults,
+    isLoaded,
+}) => {
+    const startSearchDebounced = useCallback(debounce(startSearch, 300), [
+        startSearch,
+    ]);
+    const showClearButton = hasResults && isLoaded;
+
+    return (
+        <form
+            onSubmit={(e) => {
+                e.preventDefault();
+                // Users can submit the search at any time even with no characters entered.
+                startSearch(searchQuery);
+            }}
+            className="bg bg--dark"
+            method="get"
+            role="search"
+        >
+            <div className="grid">
+                <div className="search search--programmes">
+                    <label
+                        className="search__label"
+                        htmlFor="programmes-search-input"
+                    >
+                        {label}
+                    </label>
+                    <input
+                        className="search__input input"
+                        id="programmes-search-input"
+                        type="search"
+                        placeholder={label}
+                        value={searchQuery}
+                        onChange={(e) => {
+                            const query = e.target.value;
+
+                            setQuery(query);
+
+                            if (query.length >= 3) {
+                                startSearchDebounced(query);
+                            }
+                        }}
+                    />
+                    {showClearButton ? (
+                        <button
+                            className="search__button button body body--two"
+                            type="button"
+                            onClick={(e) => {
+                                e.preventDefault();
+                                clearQuery();
```

Comment:
I think the behaviour of what that "clearQuery" does might change, but not the form UI.